students/John_Sekora/lesson05/mailroom3.py

```diff
@@ -0,0 +1,92 @@
+data = {'John Smith': [400],
+        'Bill Wilmer': [8000, 10000, 3000],
+        'George Guy': [50],
+        'Elizabeth Jones': [2000, 1000, 2000],
+        'Nathan Star': [250.50, 100]}
+
+
+def menu():
+    ''' Creates the user selection menu '''
+    while True:
+        print("\nPlease select an option:")
+        print("1. Send a Thank You   |   2. Create a Report   |   3. Send letters to everyone   |   4. Quit")
+        try:
+            menu_selection = int(input())
+        except ValueError as e:
+            print("Exception occurs in menu_selection: {}".format(e))
+            break
+        else:
+            return menu_selection
+
+
+def thank_you():
+    '''
+    Sends a Thank You: Asks for a full name, Lists the donors, Asks for donation amount,
+    Converts the donation to an integer, Adds donation amount to associated donor in list
+    '''
+    while True:
+        input_key = input("Please enter a Full Name or type 'list' to see a list of donors: ")
+        if input_key == 'list':
+            print("\nHere is a list of the current donors:\n")
+            for key, val in data.items():
+                print(f"{key:20} $  {val}")
+
+        elif input_key in data.keys():
+            while True:
+                try:
+                    input_value = float(input("Enter a donation amount for {} : ".format(input_key)))
+                except ValueError as e:
+                    print("Exception occurs in donation amount entered: {} \n".format(e))
+                    continue
```

Comment:
Don't need a continue here.  The loop will start over regardless.  `else` only runs if no exception occurs.